src/libAtomVM/externalterm.c

```diff
@@ -221,7 +223,56 @@ static void write_bytes(uint8_t *buf, avm_uint64_t val)
     }
 }
 
-static int serialize_term(uint8_t *buf, term t, GlobalContext *glb)
+static bool has_extended_utf8_encoding(const uint8_t *atom_data, size_t atom_len)
+{
+    for (size_t i = 0; i < atom_len; ) {
+        size_t out_len = 0;
+        uint32_t c;
+        enum UnicodeTransformDecodeResult res = bitstring_utf8_decode(
+            atom_data + i,
+            atom_len - i,
+            &c,
+            &out_len
+        );
+        if (res == UnicodeTransformDecodeSuccess && out_len != 1) {
+            return true;
+        } else {
+            ++i;
+        }
+    }
+    return false;
+}
+
+static inline void encode_atom_latin1(uint8_t *buf, atom_ref_t atom_ref, size_t atom_len, int *offset, GlobalContext *glb)
+{
+    *offset = 3;
+    if (!IS_NULL_PTR(buf)) {
+        buf[0] = ATOM_EXT;
+        WRITE_16_UNALIGNED(buf + 1, atom_len);
+        atom_table_write_bytes(glb->atom_table, atom_ref, atom_len, buf + 3);
+    }
+}
+
+static inline void encode_atom_utf8(uint8_t *buf, atom_ref_t atom_ref, size_t atom_len, int *offset, GlobalContext *glb)
+{
+    if (atom_len <= 255) {
```

Comment:
We don't support atoms longer than 255 bytes, so this will never be true.
So I suggest simplifying our code given this assumption.

Note: BEAM allows atoms up to 255 characters (with meaning of 255 unicode codepoints), that might be longer than 255, only at runtime. Any attempt at compiling this will fail (I suppose that's due to atom encoding in BEAM files). 

I think we should properly document this limitation.

Reply:
Yes, I have removed support for atoms longer than 255 bytes.  Well, not committed yet, but it will be.